pylanetary/spectral/core.py

```diff
@@ -0,0 +1,808 @@
+            All: Fit the spectrum with all line 5 line shape options. Data is always plotted.
+            Gaussian: fit the spectrum with a Gaussian function
+            Lorentzian: fit the spectrum with a Lorentzian function
+            Voigt: fit the spectrum with a Voigt function
+            PseudoV: fit the spectrum with a Pseudo-Voigt function
+            Moffat: fit the spectrum with a Moffat function
+            A description of all line shapes can be found at https://lmfit.github.io/lmfit-py/builtin_models.html
+    showplots : bool, optional
+          The indicator of whether to plot the best fit line shape or a comparison of line shapes. Default is True, which produces plots.
+    plot_name : string, optional
+          The name of a saved figure. Default is None, which does not save a plot.
+    initial_fit_guess: string, optional.
+          The path to a saved LMFIT result to use as the initial guess. Default is None; the LMFIT .guess method is used. 
+    
+    Returns
+    -------
+    peak : float
+          The frequency or wavelength value of the line peak, which is LMFIT's 'center' parameter.
+    err : float
+          The standard deviation of the peak frequency or wavelength value.
+    chi :
+          The calculated Chi-squared statistic of the line fit.
+    chi_red :
+          The calculated reduced Chi-squared statistic of the line fit.
+    
+    Notes
+    -----
+    Guide to keys in the nested model dictionary:
+          model[] : 
+                The Model name, taken from the user input.
+          ref : 
+                A short reference name of model type.
+          lmfit_func: 
+                The built-in LMFIT function name, needed to properly interface with the LMFIT package.
+          label :
+                The label for the legend included in the final plots.
+          color : 
+                The color of the line for a given fit type when plotting.
+    
+    An initial fit guess is not necessary, but can be very beneficial if the spectra are noisy. If your fit is failing, try including an intial guess to improve the performance of LMFIT. It is recommended that you use a pixel with as small a Doppler shift as possible to make your initial guess save file.
+    
+    '''
+    
+    if fit_type != 'All':
+      if initial_fit_guess == None:
+        pars = model[fit_type]['lmfit_func'].guess(self.spec,x=self.xaxis)
+      else:
+        inital = load_modelresult(initial_fit_guess)
+        pars = inital.params
+      fit_result = model[fit_type]['lmfit_func'].fit(self.spec,pars,x=self.xaxis,scale_covar=False)
+      print(fit_result.fit_report())
+    elif fit_type == 'All':
+      fit_result = {}
+      for key in model:
+        pars = model[key]['lmfit_func'].guess(self.spec,x=self.xaxis)
+        result = model[key]['lmfit_func'].fit(self.spec,pars,x=self.xaxis,scale_covar=False)
+        print(result.fit_report())
+        fit_result[key] = result
+    else:
```

Comment:
this is never going to be hit, because the fit_type will always either == "all" or != "all"